src/button/index.tsx

```diff
@@ -52,30 +48,29 @@ const StyledLoading = styled("button")`
 
 export const PrimaryButton: FC<
   ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>
-> = ({ text, className = "", ...props }) => {
-  return (
-    <StyledPrimary className={className} {...props}>
-      {text}
-    </StyledPrimary>
-  );
-};
+> = ({ text, loading = false, className = "", ...props }) => {
 
-export const LoadingButton: FC<
-  LoadingButtonProps & ButtonHTMLAttributes<HTMLButtonElement>
-> = ({ text, className = "", loading, ...props }) => {
   return (
-    <StyledLoading className={className} {...props}>
-      {loading && (
-        <img
-          src="https://res.cloudinary.com/dfmbuzf7l/image/upload/v1683257888/loading_fkc6bc.svg"
-          alt="loading"
-          width="25px"
-        />
-      )}
-      {text}
-    </StyledLoading>
+    <>
+      {
+        !loading ?
+          (<StyledPrimary className={className} {...props}>
+            {text}
+          </StyledPrimary>) :
+          (
+            <StyledLoading className={className} {...props}>
+              <img
+                src="https://res.cloudinary.com/dfmbuzf7l/image/upload/v1683257888/loading_fkc6bc.svg"
+                alt="loading"
+                width="25px"
+              />
```

Comment:
I would recommend only rendering the image conditionally instead of re-rendering the whole component so that runtime can be reduced!

Reply:
then we also have to remove the styles for the loading component ig because you're telling me not to use the loading component. Am i right?

Reply:
You have to remove the loading component and make a single button. I mean inside the primary button make a conditional rendering for that loading gif

Reply:
```js
const StyledPrimary = styled("button")`
  background-color: "#fff";
  border: 1px solid #000;
  border-radius: 4px;
  color: #000;
  cursor: pointer;
  font-size: 16px;
  font-weight: 600;
  padding: 8px 16px;
  transition: all 0.3s ease-in-out;
  &:hover {
    background-color: #000;
    color: #fff;
  }
  &:disabled{
    cursor: not-allowed;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 8px;
  }
`;
```

and we also have to make this in order to do so i guess..

Reply:
in order means?

Reply:
something like this 

```ts
type buttonState = 'clickable' | 'typing' | 'loading'

export interface ButtonProps {
  text?: string;
  state?: buttonState;
  className?: string;
}

export const PrimaryButton: FC<
  ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>
> = ({ text, state = 'clickable', className = "", ...props }) => {

  return (
    <>
      <StyledPrimary className={className} disabled={state === 'loading' || state === 'typing'} {...props}>
        {
          state === 'loading' ?
            <img
              src="https://res.cloudinary.com/dfmbuzf7l/image/upload/v1683257888/loading_fkc6bc.svg"
              alt="loading"
              width="25px"
            /> : ''
        }

        {text}
      </StyledPrimary>
    </>
  );
}

```

by "in order" i meant to combine both primary btn and loading btn into a single component XD, sorry for that.

Reply:
just have a loading state, no need for multiple states! 
disable if loading is true and if loading is true show the image.
does this make sense?

Reply:
for form validation a guy need multiple states i guess.
for form validation they also require a disabled state i.e typing one and if filled they have to make the button clickable.

Reply:
Okay! Cool

Reply:
you can do that

Reply:
ok let me push : )